src/ripple/net/impl/RPCCall.cpp

```diff
@@ -495,6 +495,48 @@ class RPCParser
         return rpcError (rpcINVALID_PARAMS);
     }
 
+    bool isValidJson2(Json::Value const& jv)
+    {
+        if (jv.isObject())
+        {
+            if (jv.isMember("jsonrpc") && jv["jsonrpc"] == "2.0" &&
+                jv.isMember("ripplerpc") && jv["ripplerpc"] == "2.0" &&
+                jv.isMember("id") && jv.isMember("method"))
+                return true;
+        }
+        return false;
+    }
+
+    Json::Value parseJson2(Json::Value const& jvParams)
+    {
+        Json::Reader reader;
+        Json::Value jv;
+        reader.parse(jvParams[0u].asString(), jv);
+        if (isValidJson2(jv))
+        {
+            Json::Value jv1{Json::objectValue};
+            jv1["jsonrpc"] = jv["jsonrpc"];
+            jv1["ripplerpc"] = jv["ripplerpc"];
+            jv1["id"] = jv["id"];
+            jv1["method"] = jv["method"];
+            if (jv.isMember("params"))
+            {
+                auto const& params = jv["params"][0u];
```

Comment:
This will `Throw` if `params` isn't an array. Do we catch from all possible paths?